### DISCUSSION TITLE
Home key does not work but live tv does?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [WesSouza/lgtv-ip-control#110] [Q&A] [answered]
This works...

TV.connect()
        .then(async () => {
                await TV.sendKey('livetv')

                process.exit()
        })
.catch(console.error);

But this does not work...

TV.connect()
        .then(async () => {
                await TV.sendKey('home')

                process.exit()
        })
.catch(console.error);

AssertionError [ERR_ASSERTION]: key must be valid
    at LGTV.sendKey (/root/node_modules/lgtv-ip-control/dist/classes/LGTV.cjs:77:5)
    at /root/LGTV-Screen-Off.js:6:12 {
  generatedMessage: false,
  code: 'ERR_ASSERTION',
  actual: false,
  expected: true,
  operator: '=='
}

The home key is defined in the keys enumeration, so not sure why it is not a valid key.

**Comment by Jibun-no-Kage** [comment]
How do you reference and exported enum?

const { LGTV } = require('lgtv-ip-control');
const TV = new LGTV('x.x.x.x', 'xx:xx:xx:xx:xx:xx', XXXXXXXX');

Object.keys(Keys).forEach(key => console.log("Key:", key))

Object.keys(Keys).forEach(key => console.log("Key:", key))
            ^

ReferenceError: Keys is not defined
    at Object.<anonymous> (/root/LGTV-Keys.js:4:13)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12)
    at node:internal/main/run_main_module:28:49

**Comment by Jibun-no-Kage** [comment]
Figured it out... the Key 'home' is NOT 'home' but 'myapp'... So your documentation (read me) is in error 'home' = 'myapp'  NOT 'home'='home'.  Guess the key should be named 'myapp' to be descriptive as to its true value?  However, at least on my LG TV, 'myapp' returns "This button is not available", so not sure what string value will actually engage the Home button?

**Comment by WesSouza** [comment, marked as answer]
This is a common practice with JavaScript constants, where the property name has a "human readable" value, and the property value has the actual expected machine value.

So basically you have the convenience name that resolves to the actual value:

```js
const { LGTV, Apps } = require("lgtv-ip-control");

console.log(Apps.youtube);
// Resolves to "youtube.leanback.v4"
```

That's why `Apps`, `Inputs`, etc. are exported for you to use.

**Comment by Jibun-no-Kage** [comment]
Yeah, I am or was having Node Red reference the exported keys lists... but your example illustrates how to do it.  Still learning JavaScript. or should I say TypeScript.
